app/src/test/java/org/mozilla/fenix/settings/SettingsFragmentTest.kt

```diff
@@ -116,6 +119,35 @@ class SettingsFragmentTest {
         assertTrue(preferenceAmoCollectionOverride.isVisible)
     }
 
+    @Test
+    fun `Add-on collection override pref is not visible if feature is disabled`() = runTestOnMain {
+        val settingsFragment = SettingsFragment()
+        val activity = Robolectric.buildActivity(FragmentActivity::class.java).create().get()
+
+        activity.supportFragmentManager.beginTransaction()
+            .add(settingsFragment, "test")
+            .commitNow()
+
+        advanceUntilIdle()
+
+        mockkObject(FeatureFlags)
+        every { FeatureFlags.customExtensionCollectionFeature } returns false
+
+        val preferenceAmoCollectionOverride = settingsFragment.findPreference<Preference>(
+            settingsFragment.getPreferenceKey(R.string.pref_key_override_amo_collection),
+        )
+
+        val settings: Settings = mockk(relaxed = true)
+        settingsFragment.setupAmoCollectionOverridePreference(settings)
+        assertNotNull(preferenceAmoCollectionOverride)
+        assertFalse(preferenceAmoCollectionOverride!!.isVisible)
+
+        every { settings.showSecretDebugMenuThisSession } returns true
+        every { settings.amoCollectionOverrideConfigured() } returns true
+        settingsFragment.setupAmoCollectionOverridePreference(settings)
+        assertFalse(preferenceAmoCollectionOverride.isVisible)
+    }
+
```

Comment:
The visibility of the feature depends on both the feature flag and either the flag for showing the debug menu or whenever the collection override has already been  configured. Instead of hard-coding the release channel in the Setup func of the test class, the feature flag is introduced to the test functions to address the enabling of the feature in a more flexible way.